rust-bins/src/bin/keygen-genesis.rs

```diff
@@ -0,0 +1,241 @@
+use clap::AppSettings;
+use client_server_helpers::*;
+use crypto_common::*;
+use curve_arithmetic::Curve;
+use elgamal::PublicKey;
+use id::types::*;
+use std::convert::TryFrom;
+
+use pairing::bls12_381::Bls12;
+use sha2::Sha512;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use curve25519_dalek::edwards::CompressedEdwardsY;
+use ed25519_dalek::*;
+use pairing::bls12_381::{G1, G2};
+use std::fs;
+#[derive(StructOpt)]
+struct KeygenIp {
+    #[structopt(long = "rand-input", help = "File with randomness.")]
```

Comment:
Maybe it would be better to call this something else. The argument is not really randomness. Maybe seed.